platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/datasource/VirtualDataSource.java

```diff
@@ -223,4 +180,8 @@ default void stopBackgroundCompaction() {}
      * @return a new key set
      */
     VirtualKeySet<K> buildKeySet();
+
+    long getFirstLeafPath();
+
+    long getLastLeafPath();
```

Comment:
Are these two new method really used at VirtualMap level? The state (and first/last leaf paths) is stored in VirtualMap, why would a data source be ever queried for it?

Reply:
These methods are used in `VirtualRootNode#fullLeafRehash` to get a valid range of leaves to process . Given the fact that I need to know the range for the data source specifically, I thought that it's the most straightforward way of doing it.